src/main/java/sirius/web/controller/ControllerDispatcher.java

```diff
@@ -203,15 +204,17 @@ private void performRoute(WebContext ctx, Route route, List<Object> params) {
 
     private void executeRoute(WebContext ctx, Route route, List<Object> params) throws Exception {
         ctx.setAttribute(ATTRIBUTE_MATCHED_ROUTE, route.getPattern());
-        if (route.isJSONCall()) {
-            executeJSONCall(ctx, route, params);
+        if (route.getApiResponseFormat() == Format.JSON) {
+            executeApiCall(ctx, route, ctx.respondWith().json(), params);
+        } else if (route.getApiResponseFormat() == Format.XML) {
+            executeApiCall(ctx, route, ctx.respondWith().xml(), params);
         } else {
             route.invoke(params);
         }
     }
 
-    private void executeJSONCall(WebContext ctx, Route route, List<Object> params) throws Exception {
-        JSONStructuredOutput out = ctx.respondWith().json();
+    private void executeApiCall(WebContext ctx, Route route, StructuredOutput out, List<Object> params)
```

Comment:
Question: Why is the strcutured output created outside and passed in instead of using the route.getApiResponseFormat inside this function to determine which output should be created?